tfs/reader.py

```diff
@@ -277,15 +311,40 @@ def _read_metadata(tfs_file_path: pathlib.Path | str) -> _TfsMetaData:
     )
 
 
-def _parse_header(str_list: list[str]) -> tuple:
+def _parse_header(str_list: list[str]) -> tuple[str, bool | str | int | float]:
+    """
+    Expects a parsed header line content after the '@' header identifier,
+    as a .
+
+    Args:
+        str_list (list[str]): list of parsed elements from the header line
+            (we use parse in reader with 'shlex.split`).
+
+    Returns:
+        A tuple with the name of the header parameter for this line, as
+        well as its value cast to the proper type (as determined by the
+        type identifier).
+
+    Raises:
+        AbsentTypeIdentifierError: if no type identifier is found in the header line.
+        InvalidBooleanHeaderError: if the identifier type indicates a boolean
+            but the corresponding value is not an accepted boolean.
+    """
     type_index = next((index for index, part in enumerate(str_list) if part.startswith("%")), None)
     if type_index is None:
-        errmsg = f"No data type found in header: '{''.join(str_list)}'"
-        raise TfsFormatError(errmsg)
+        raise AbsentTypeIdentifierError(str_list)
+
+    # Get name and string of the header, and determine its type
+    name: str = " ".join(str_list[0:type_index])
+    value_string: str = " ".join(str_list[(type_index + 1) :])
+    value_string: str = value_string.strip('"')
+    value_type: type = _id_to_type(str_list[type_index])
 
-    name = " ".join(str_list[0:type_index])
-    value_string = " ".join(str_list[(type_index + 1) :])
-    return name, _id_to_type(str_list[type_index])(value_string.strip('"'))
+    if value_type is bool:  # special handling for boolean values
+        return name, _string_to_bool(value_string)
+    elif value_type is np.complex128:  # MAD-NG uses i for imaginary but Python needs j
+        value_string = value_string.replace("i", "j")
```

Comment:
**FEATURE**

We have a special handling of complex values as `MAD-NG` writes them with `i` and not `j` for the complex part.

Note that if `j` was used in the file somehow we are still parsing correctly.